### PR TITLE
ST-09031: Extract Tool Registry Registration and Mutation Paths

### PR DESCRIPTION
**Pull request by TVScoundrel**
## Story
- ST-09031: Extract Tool Registry Registration and Mutation Paths

## What Changed
- extracted register, remove, update, bulk-registration, and clear logic from `packages/core/src/tools/registry.ts` into `packages/core/src/tools/registry-mutations.ts`
- kept `ToolRegistry` as the stable public facade while delegating the mutation paths to focused internal helpers
- added focused helper coverage in `packages/core/tests/tools/registry-mutations.test.ts`
- updated story tracking and documentation for the extraction and validation results

## Acceptance Criteria Coverage
- `packages/core/src/tools/registry.ts` now delegates registration, update, removal, and bulk-registration logic into focused helpers while keeping the public entrypoint stable
- public duplicate detection, name consistency checks, and emitted event behavior are preserved
- focused tests cover registration conflicts, updates, removals, clear semantics, and bulk-registration edge cases
- explicit-`any` warning deltas were recorded in `docs/st09031-tool-registry-registration-mutation-extraction.md` with no regressions in touched files
- story documentation was added at `docs/st09031-tool-registry-registration-mutation-extraction.md`

## Validation Performed
- `pnpm exec tsc -p packages/core/tsconfig.json --noEmit`
- `pnpm exec eslint packages/core/src/tools/registry.ts packages/core/src/tools/registry-mutations.ts packages/core/tests/tools/registry.test.ts packages/core/tests/tools/registry-mutations.test.ts`
  - passed cleanly
- `pnpm test --run packages/core/tests/tools/registry.test.ts packages/core/tests/tools/registry-events.test.ts packages/core/tests/tools/registry-collection.test.ts packages/core/tests/tools/registry-prompt.test.ts packages/core/tests/tools/registry-mutations.test.ts`
  - `5 passed` files, `59 passed` tests
- `pnpm lint:explicit-any:baseline --silent`
  - `180/289` warnings, `core 63/119`
- `pnpm test --run`
  - `162 passed | 16 skipped` files
  - `2225 passed | 286 skipped` tests
- `pnpm lint`
  - exit `0`; warnings only

## Status
- Ready for review

### CODE REVIEW

File: docs/st09031-tool-registry-registration-mutation-extraction.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+`packages/core/src/tools/registry.ts` still carried the public mutation paths after the earlier collection, prompt, and event helper splits. This story extracted registration, update, removal, bulk-registration, and clear behavior into a focused internal helper so the public `ToolRegistry` facade stays stable while the mutation boundary becomes easier to review and evolve.
+
+## What Changed
+
+| File | Change |
+|------|--------|
+| `packages/core/src/tools/registry.ts` | Reduced the public registry facade to delegate mutation paths to focused helpers while preserving the existing public API and event flow. |
+| `packages/core/src/tools/registry-mutations.ts` | Added focused helper functions for register, remove, update, bulk register, and clear operations, preserving the existing duplicate/conflict and name-consistency error messages. |
+| `packages/core/tests/tools/registry-mutations.test.ts` | Added focused helper coverage for registration conflicts, updates, removals, clear behavior, and bulk-registration edge cases. |
```
[the remark this answers is not in the document]
Comment (TVScoundrel) [reply docs/st09031-tool-registry-registration-mutation-extraction.md:13]: Checked directly. The table is already valid markdown with a single leading `|` on each row, so no doc change is needed here. I’m resolving this as a false positive.
